src/rust/ide/src/controller/searcher.rs

```diff
@@ -957,6 +1187,34 @@ mod test {
         assert!(Rc::ptr_eq(&arg.picked_suggestion,&entry2));
     }
 
+    #[test]
```

Comment:
this whole test is total TODO, I've just started it.
Basically I expect to finish commented out cases, add more checks and more expected method calls, as needed.